From f9695a61fd7b5fa357f5bf0dc2d4e0794536f50d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Mon, 22 Apr 2024 15:15:05 +0300
Subject: [PATCH] Disable code path not supported by MPIR

---
 src/gmpy2_pow.c | 80 +------------------------------------------------
 1 file changed, 1 insertion(+), 79 deletions(-)

diff --git a/src/gmpy2_pow.c b/src/gmpy2_pow.c
index fca7af9..1d57385 100644
--- a/src/gmpy2_pow.c
+++ b/src/gmpy2_pow.c
@@ -682,85 +682,7 @@ PyDoc_STRVAR(GMPy_doc_integer_powmod_sec,
 static PyObject *
 GMPy_Integer_PowMod_Sec(PyObject *self, PyObject *args)
 {
-    PyObject *x, *y, *m;
-    int xtype, ytype, mtype;
-    MPZ_Object *tempx = NULL, *tempy = NULL, *tempm = NULL, *result = NULL;
-    CTXT_Object *context = NULL;
-
-    CHECK_CONTEXT(context);
-
-    if (PyTuple_GET_SIZE(args) != 3) {
-        TYPE_ERROR("powmod_sec() requires 3 arguments.");
-        goto err;
-    }
-
-    if (!(result = GMPy_MPZ_New(NULL))) {
-        goto err;
-    }
-
-    x = PyTuple_GET_ITEM(args, 0);
-    y = PyTuple_GET_ITEM(args, 1);
-    m = PyTuple_GET_ITEM(args, 2);
-
-    xtype = GMPy_ObjectType(x);
-    ytype = GMPy_ObjectType(y);
-    mtype = GMPy_ObjectType(m);
-    
-    /* Validate base. */
-
-    if (!IS_TYPE_INTEGER(xtype)) {
-        TYPE_ERROR("powmod_sec() base must be an integer.");
-        goto err;
-    }
-
-    if (!(tempx = GMPy_MPZ_From_IntegerWithType(x, xtype, NULL))) {
-        goto err;
-    }
-
-    /* Validate exponent. It must be > 0. */
-
-    if (!IS_TYPE_INTEGER(ytype)) {
-        TYPE_ERROR("powmod_sec() exponent must be an integer.");
-        goto err;
-    }
-
-    if (!(tempy = GMPy_MPZ_From_IntegerWithType(y, ytype, NULL))) {
-        goto err;
-    }
-
-    if (!(mpz_sgn(tempy->z) == 1)) {
-        VALUE_ERROR("powmod_sec() exponent must be > 0.");
-        goto err;
-    }
-    /* Validate modulus. It must be odd.*/
-
-    if (!IS_TYPE_INTEGER(mtype)) {
-        TYPE_ERROR("powmod_sec() modulus must be an integer.");
-        goto err;
-    }
-
-    if (!(tempm = GMPy_MPZ_From_IntegerWithType(m, mtype, NULL))) {
-        goto err;
-    }
-
-    if (mpz_even_p(tempm->z)) {
-        VALUE_ERROR("powmod_sec() modulus must be odd.");
-        goto err;
-    }
-
-    GMPY_MAYBE_BEGIN_ALLOW_THREADS(context);
-    mpz_powm_sec(result->z, tempx->z, tempy->z, tempm->z);
-    GMPY_MAYBE_END_ALLOW_THREADS(context);
-    
-    Py_DECREF(tempx);
-    Py_DECREF(tempy);
-    Py_DECREF(tempm);
-    return (PyObject*)result;
-
-  err:
-    Py_XDECREF(tempx);
-    Py_XDECREF(tempy);
-    Py_XDECREF(tempm);
+    RUNTIME_ERROR("MPIR does not support mpz_powm_sec");
     return NULL;
 }
 
-- 
2.44.0

